### PR TITLE
Build and run improvements

### PR DESCRIPTION
**Pull request by Naghasan**
- Add an LLVM-MLIR profile
- Enable queue profiling to track kernel time
- Report individual time sample rather than a string to parse
- Add an actual command line parsing for run-suite

### CODE REVIEW

**Review by whitneywhtsang — APPROVED** [review]

File: bin/run-suite [hunk 1]
```diff
@@ -6,6 +6,54 @@ import subprocess
 import sys
 import copy
 import timeit
+import io
+import argparse
+
+app_list = [
```
Comment (sommerlukas) [inline bin/run-suite]: Would it make sense to create a more focused list/a different profile for the benchmarks we're currently interested in?

File: CMakeLists.txt [hunk 2]
```diff
@@ -42,10 +43,13 @@ if(SYCL_IMPL STREQUAL "ComputeCpp")
 elseif(SYCL_IMPL STREQUAL "hipSYCL")
   find_package(hipSYCL CONFIG REQUIRED)
 elseif(SYCL_IMPL STREQUAL "LLVM")
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsycl -fsycl-targets=spir64-unknown-unknown-syclmlir -w")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsycl")
```
Comment (sommerlukas) [inline CMakeLists.txt]: Should we add the flag for omitting the rounded range kernel here and below?

**Review by sommerlukas — COMMENTED** [review]
> Report individual time sample rather than a string to parse

Just to clarify: This is currently not used by the visualization scripts, as in Python it is more comfortable to parse the string than to check for presence/absence of columns in the CSV.

**Comment by Naghasan** [comment]
> Just to clarify: This is currently not used by the visualization scripts, as in Python it is more comfortable to parse the string than to check for presence/absence of columns in the CSV.

I'll probably remove that then

File: CMakeLists.txt (hunk 2 above)
Comment (Naghasan) [reply CMakeLists.txt]: that's a very good point, +1

@etiotto @whitneywhtsang WDYT ?

File: bin/run-suite (hunk 1 above)
Comment (Naghasan) [reply bin/run-suite]: Added the option `--no-bandwidth-test` which excludes the bandwidth we don't care about

**Comment by Naghasan** [comment]
I added a split option `per_kernel`, that should allow us to run the fp32 version of some kernels on devices that don't support fp64. As both version are defined in the same TU, the whole SPIR-V module is flagged as using fp64 and so the driver will reject it regardless of the kernel we want to use. With per_kernel each kernel is in his own module so the driver shouldn't reject the fp32 variants.

File: CMakeLists.txt (hunk 2 above)
Comment (whitneywhtsang) [reply CMakeLists.txt]: As discussed, we agree to put `-Xclang -fsycl-disable-range-rounding`

File: bin/run-suite [hunk 3]
```diff
@@ -6,6 +6,65 @@ import subprocess
...
+
+is_bandwidth_bench = {
+      "arith",
+      "blocked_transform",
+      "dag_task_throughput_independent",
+      "dag_task_throughput_sequential",
+      "DRAM",
+      "host_device_bandwidth",
+      "local_mem",
+      #"matmulchain",
```
Comment (whitneywhtsang) [inline bin/run-suite]: Why is it commented out? is it a bandwidth bench or not?

File: bin/run-suite [hunk 4]
```diff
@@ -37,14 +96,43 @@ def create_log_range(begin, end):
...
+
+parser = argparse.ArgumentParser(description="Run benchmarks",
+                                 formatter_class=argparse.RawTextHelpFormatter)
+parser.add_argument("profile", metavar="profile", choices={'default', 'quicktest', 'cpu', 'gpu', 'cpu-noverify', 'cpu-nondrange', 'cpu-noverify-nondrange', 'gpu-noverify'})
+parser.add_argument("-log", metavar="outfile",
+                    default=sys.stderr,
+                    type=argparse.FileType('w', encoding='UTF-8'),
+                    help="Log file")
+parser.add_argument("--try-fail", action='store_true', help="Try known failures")
+parser.add_argument("--no-bandwidth-test", action='store_true', help="Exclude known failures")
```
Comment (whitneywhtsang) [inline bin/run-suite]:
```suggestion
parser.add_argument("--no-bandwidth-test", action='store_true', help="Exclude bandwidth tests")
```
?

File: bin/run-suite (hunk 3 above)
Comment (Naghasan) [reply bin/run-suite]: Could have been of interest, fixed

File: bin/run-suite (hunk 4 above)
Comment (Naghasan) [reply bin/run-suite]: updated the comment

**Review by whitneywhtsang — APPROVED** [review]

**Comment by whitneywhtsang** [comment]
Can we merge it or need to wait for 2 approvals?

**Comment by Naghasan** [comment]
I think we are good @sommerlukas did a review, we can post-fix things if needed.

**Pull request merged**